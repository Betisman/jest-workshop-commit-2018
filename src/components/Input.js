import React from 'react';

const Input = ({ onChange, required, type, placeholder, value }) => 
  <input
    onChange={onChange}
    required={required}
    type={type}
    placeholder={placeholder}
    value={value}
  />;

export default Input;
