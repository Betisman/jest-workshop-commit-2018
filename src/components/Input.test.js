import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './Input';

const props = {
  onChange: () => 0,
};

describe('<Input />', function() {
  it('Match snapshot', () => {
    const tree = renderer.create(<Input {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onChangeProps with right values', () => {
    const mockProps = {
      ...props,
      onChange: jest.fn(),
    };
    const wrapper = shallow(<Input {...mockProps} />);
    wrapper.find('input').props().onChange('change');
    expect(mockProps.onChange).toHaveBeenCalledWith('change');
  });
});
