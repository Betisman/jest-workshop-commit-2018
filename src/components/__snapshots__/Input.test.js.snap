// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Match snapshot 1`] = `
<input
  onChange={[Function]}
/>
`;
